### ISSUE TITLE
Support i18n (internationalization)

### ISSUE THREAD

**Issue by JarvyJ** [JarvyJ/HomeIntent#162]
I was initially planning to go with a JSON based approach, but after chatting with an i18n consultant - I've decided to go with a "DSL-style" approach (except that the language is actually Python).

The general idea is that not all languages can easily generate sentences for all the different intents (does "toggle" always translate?) and some languages may require specific modifications that may need "pre-processing" before being handled. As such, the rough plan is to separate out the sentences from the handling functions.

There will be a set of "core" handling methods (the actual code to turn lights on, off, toggle, etc) and then the language code files that call the core methods. The language files will handle the language's sentences, do any pre-processing, call the "core" method, and then return in the language. 

Hopefully this will work out well. I want to keep the "core" methods and language stuff in the same spot for easy reference. I'm a little worried about the timer method, but that has some other issues as well that will crop up with the addition of satellites, so it might be good to rework it as needed.

**Comment by JarvyJ** [comment]
work on this has started in [a new branch](https://github.com/JarvyJ/HomeIntent/tree/homeintent/i18n). Currently there are a couple of breaking changes:

 * when the [`setup`](https://homeintent.io/reference/developer/example-component/#setup) function is called, Home Intent will pass in the selected language. This is currently required, so existing custom components will break...
 * all new "first-party" components will have to support (as in the ability to) i18n moving forward. custom components will at the minimum have to take in the parameter, but don't have to do anything with it.

backwards compatible changes

 * the [`get_file`](https://homeintent.io/reference/api/get-file/) method now has a `language` parameter that will load files from `default_configs/{language}` if provided.

After converting over all the files, basically anything that is language specific will get moved to the language file (ex `light/en.py`) and anything common can just live in the base file (ex `light/base_light.py`)

We still need to document the changes to the above functions, the settings change, and how to translate the files. Along with that, we need to get "fallback" language support in. As of right now everything has to be translated to work in another language. We should probably have it fallback to english if the functionality isn't there for another language.

**Comment by JarvyJ** [comment]
first part is in with #175!

**Comment by JarvyJ** [comment]
I've managed to undo the breaking change and instead just make the existing `get_file` to be considered deprecated. It's essentially just been moved into the `home_intent` class that is initialized and passed around. It now by default pulls the language specific version of a file and then falls back to english if it doesn't work.

The only other major change is we've added an `import_module` into Home Intent which can be used to dynamically load classes. This is also language aware, so all first party components will have to use it to load the language specific version of their library.

With these changes, the modifications to setup are no longer required. Technically any written custom components should just continue to work. We've added a deprecation notice to the package level `get_file` to use the object level version of it instead.
